web/src/buddy_data.ts

```diff
@@ -312,10 +311,9 @@ function filter_user_ids(user_filter_text: string, user_ids: number[]): number[]
     // This first filter is for whether the user is eligible to be
     // displayed in the right sidebar at all.
     user_ids = user_ids.filter((user_id) => {
-        const person = people.maybe_get_user_by_id(user_id);
+        const person = people.maybe_get_user_by_id(user_id, true);
 
         if (!person) {
-            blueslip.warn("Got user_id in presence but not people: " + user_id);
```

Comment:
Maybe a comment of the form:

// See the comments in presence.set_info for details, but this is an expected race. User IDs for whom we have presence but no user metadata should be skipped.